apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTextBuilderTests.cs

```diff
@@ -77,6 +77,23 @@ public void DropDatabaseUnrecognized(string ddlString)
         [InlineData("\t\nWITH AlbumsView AS (SELECT * FROM Albums) select Title from AlbumsView")]
         [InlineData(" WITH   AlbumsView AS (SELECT * FROM Albums) select Title from AlbumsView")]
         [InlineData("WITH\t\nAlbumsView AS (SELECT * FROM Albums) select Title from AlbumsView")]
+        [InlineData("-- Single line comment\nSELECT * FROM Albums")]
+        [InlineData("# Single line comment\nSELECT * FROM Albums")]
+        [InlineData("/* Multi\nline\ncomment\n */\nSELECT * FROM Albums")]
+        [InlineData("/* Multi\n * line\n * comment\n */\nSELECT * FROM Albums")]
+        [InlineData("@{FORCE_INDEX=_BASE_TABLE} SELECT * FROM Albums")]
+        [InlineData("@{FORCE_INDEX=_BASE_TABLE} \n-- Single line comment\nSELECT * FROM Albums")]
+        [InlineData("@{FORCE_INDEX=_BASE_TABLE} \n# Single line comment\nSELECT * FROM Albums")]
+        [InlineData("@{FORCE_INDEX=_BASE_TABLE} /* Multi\nline\ncomment\n */\nSELECT * FROM Albums")]
+        [InlineData("@{FORCE_INDEX=_BASE_TABLE} /* Multi\n * line\n * comment\n */\nSELECT * FROM Albums")]
+        [InlineData("-- Single line comment\n@{FORCE_INDEX=_BASE_TABLE} \nSELECT * FROM Albums")]
+        [InlineData("# Single line comment\n@{FORCE_INDEX=_BASE_TABLE} \nSELECT * FROM Albums")]
+        [InlineData("/* Multi\nline\ncomment\n */\n@{FORCE_INDEX=_BASE_TABLE} \nSELECT * FROM Albums")]
+        [InlineData("/* Multi\n * line\n * comment\n */\n@{FORCE_INDEX=_BASE_TABLE} \nSELECT * FROM Albums")]
+        [InlineData("-- Single line comment\n@{FORCE_INDEX=_BASE_TABLE, OPTIMIZER_VERSION=1} \n\t \nSELECT * FROM Albums")]
+        [InlineData("-- Single line comment\n@{FORCE_INDEX=_BASE_TABLE, OPTIMIZER_VERSION=1} -- Another single line comment \n\t @{FORCE_INDEX=_BASE_TABLE, OPTIMIZER_VERSION=1} /* Multi\n * line\n * comment\n */\nSELECT * FROM Albums")]
+        [InlineData("-- Single line comment\n@{FORCE_INDEX=_BASE_TABLE, /* Multi\n * line\n * comment\n */ OPTIMIZER_VERSION=1} \n\t \nSELECT * FROM Albums")]
+        [InlineData("-- Single line comment\n@{FORCE_INDEX=_BASE_TABLE, -- Inner single line comment\n OPTIMIZER_VERSION=1} \n\t \nSELECT * FROM Albums")]
         public void SelectCommand(string commandText)
```

Comment:
The following valid statements will fail, but it should be easily fixable in the check that searches for the keyword.

```suggestion
        [InlineData("SELECT/*comment*/ * FROM Albums")]
        [InlineData("SELECT--comment\n * FROM Albums")]
        public void SelectCommand(string commandText)
```

Reply:
Yikes with those being valid! Funnily enough, without this PR, something like `SELECT/* comment */ * FROM myTable` will fail, but `SELECT /* comment */ * FROM myTable` will not.

I think the easiest way to account for this is to split the stripped command on `/` and `-` and `#` as well as blanks. WDYT?

Reply:
> I think the easiest way to account for this is to split the stripped command on / and - and # as well as blanks. WDYT?

That, or just checking whether the command string starts with one of the known keywords. That would mean that a statement that starts with `SELECTINSERT` would be seen as a query (instead of being rejected by the client library), and then be handled by the backend.